lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -1291,6 +1410,11 @@ class CopyForwardingPass : public SILFunctionTransform
         }
       }
 
+    // Eliminate short-lived temporaries.
+    for (auto &rvalue : RValueCopies) {
+      removeTempRValue(rvalue);
```

Comment:
Is it always guaranteed that the instructions which are deleted here are not referenced in the subsequent code?
E.g. if there are 2 copy_addrs referencing the same alloc_stack

Maybe it's saver that removeTempRValue just does a replaceAllUsesWith and let the alloc/dealloc/destroy/copy_addr be deleted by a subsequent cleanup pass.
This would also simplify the code a bit, because you would not need the TempRValue struct.